arviz/data/inference_data.py

```diff
@@ -173,6 +175,53 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def extend_xr_method(func):  # pylint: disable=no-self-argument
+        """Wrapper to add methods to InferenceData Class"""
+
+        @functools.wraps(func)
```

Comment:
what does this do?

Reply:
This is used to create wrapper function which wraps around func. You can check it out [here](https://docs.python.org/3/library/functools.html#functools.wraps).